app/controllers/my_notifications.js

```diff
@@ -1,50 +1,83 @@
   }),
 
-  model: Ember.computed("allMessages.@each.state", "session.currentUser.id", "allMessages.@each.offer.createdBy", function(){
-    var currentUserId = this.get('session.currentUser.id');
+  model: Ember.computed(
+    "allMessages.@each.state",
+    "session.currentUser.id",
+    "allMessages.@each.offer.createdBy",
+    function() {
+      var currentUserId = this.get("session.currentUser.id");
 
-    return this.get("allMessages").rejectBy("state", "never-subscribed").rejectBy("offer.createdBy.id", currentUserId);
-  }),
+      return this.get("allMessages")
+        .rejectBy("state", "never-subscribed")
+        .rejectBy("offer.createdBy.id", currentUserId);
+    }
+  ),
 
+  hasLoadedReadMessages: false,
   showUnread: Ember.computed({
     get: function() {
-      return false;
+      return true;
     },
     set: function(key, value) {
       return value;
     }
   }),
 
-  myNotifications: Ember.computed('showUnread', 'allNotifications', function(){
-    return this.get('showUnread') ? this.get('unreadNotifications') : this.get('allNotifications');
+  myNotifications: Ember.computed("showUnread", "allNotifications", function() {
+    if (this.get("showUnread")) {
+      return this.get("unreadNotifications");
+    }
+    return this.get("allNotifications");
+  }),
+
+  unreadNotifications: Ember.computed("allNotifications.[]", function() {
+    return this.get("allNotifications").rejectBy("unreadCount", 0);
   }),
 
-  unreadNotifications: Ember.computed('allNotifications.[]', function(){
-    return this.get('allNotifications').rejectBy('unreadCount', 0);
+  readNotifications: Ember.computed("allNotifications.[]", function() {
+    return this.get("allNotifications").filterBy("unreadCount", 0);
   }),
 
-  allNotifications: Ember.computed("model.@each.state", function(){
+  allNotifications: Ember.computed("model.@each.state", function() {
     var keys = {};
     var res = [];
-    this.get("sortedModel").forEach(function(message) {
+    this.get("sortedModel").forEach(message => {
       var isPrivate = message.get("isPrivate");
-      var key = isPrivate + message.get("offer.id") + (message.get("item.id") || "");
+      var key =
+        isPrivate + message.get("offer.id") + (message.get("itemId") || "");
       if (!keys[key]) {
-        var props = ["id", "item", "offer", "sender", "createdAt", "isPrivate"];
+        var props = [
+          "id",
+          "itemId",
+          "offer",
+          "sender",
+          "createdAt",
+          "isPrivate"
+        ];
         var notification = Ember.Object.create(message.getProperties(props));
-        notification.set("unreadCount", message.get("state") === "unread" ? 1 : 0);
+        notification.set(
+          "unreadCount",
+          message.get("state") === "unread" ? 1 : 0
+        );
         notification.set("text", message.get("body"));
         notification.set("isSingleMessage", message.get("state") === "unread");
+        if (notification.get("itemId")) {
```

Comment:
store only if present. The use is exclusively for the item image on the notification list. May result in not having images sometimes, but we discussed it and that's worth the speed improvement

Reply:
Great.